src/Configuration.php

```diff
@@ -129,18 +154,45 @@ public function getDOMDocument() {
      * @param  string     $xpathExpression
      * @param  boolean    $alwaysReturnArray
      * @return array|null
-     **/
+     * */
     public function get($xpathExpression, $alwaysReturnArray = false) {
+        if ($this->useCaching && $alwaysReturnArray) {
+            if (!array_key_exists($xpathExpression, $this->cache["alwaysArray"])) {
```

Comment:
How do you propose the value can be wrapped on the fly considering this code fragment (I changed the getFromDOMDocument so it always returns an unwrapped array which could be cached and wrapped on the fly):

    private function getFromDOMDocument($xpathExpression)
    {
        if (($this->xpath instanceof DOMXPath) === false) {
            return;
        }

        $nodeList = @$this->xpath->query($xpathExpression);
        if ($nodeList instanceof DOMNodeList) {
            if ($nodeList->length === 1) {
                return $this->nodeToArray($nodeList->item(0)); // this is an unwrapped array that'd require wrapping
            } else {
                $result = array();
                foreach ($nodeList as $node) {
                    $result[] = $this->nodeToArray($node);
                }

                return $result; // this is an unwrapped array that doesn't require wrapping
            }
        } 
    }

Similar, if you'd let getFromDOMDocument wrap the result in arrays as it did, you can't know in your cache if you're dealing with a wrapped or unwrapped array because of:

    if ($nodeList->length === 1) {
        if ($alwaysReturnArray === true) {
           return array($this->nodeToArray($nodeList->item(0)));
    }

    return $this->nodeToArray($nodeList->item(0));

Both returns give an array